Sample.App.Database/Program.cs

```diff
@@ -1,21 +1,65 @@
-using Beef.Database.Core;
-using System.Threading.Tasks;
-
-namespace Sample.App.Database
-{
-    /// <summary>
-    /// Represents the <b>database utilities</b> program (capability).
-    /// </summary>
-    public static class Program
-    {
-        /// <summary>
-        /// Main startup.
-        /// </summary>
-        /// <param name="args">The startup arguments.</param>
-        /// <returns>The status code whereby zero indicates success.</returns>
-        static Task<int> Main(string[] args)
-        {
-            return DatabaseConsoleWrapper.Create("Data Source=.;Initial Catalog=SampleDB;Integrated Security=True", "Sample", "App").RunAsync(args);
-        }
-    }
-}
\ No newline at end of file
+﻿using System;
+using System.Data.SqlClient;
+using System.Threading;
+using DbUp;
+
+namespace Sample.App.Database
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.Error.WriteLine($"Expecting a single connection string argument but received {args.Length} argument(s).");
+                Environment.Exit(-1);
+                return;
+            }
+
+            var connectionString = args[0];
+
+            ExecuteWithRetry(
+                () => EnsureDatabase.For.SqlDatabase(connectionString),
+                retryCondition: ex => ex is SqlException sqlEx && (sqlEx.Number == 17187 || sqlEx.Number == 18456),
+                onRetry: (ex, attempts, maxTimes, delay) => Console.WriteLine($"SQL Server is not ready. Retry {attempts}/{maxTimes} in {delay}ms"),
+                maxAttempts: 10,
+                delay: 1000);
+
+
+            var dbup = DeployChanges.To
+                .SqlDatabase(connectionString)
+                .LogToConsole()
+                .LogScriptOutput()
+                .WithExecutionTimeout(TimeSpan.FromSeconds(30))
+                .WithScriptsFromFileSystem("./scripts")
+                .Build();
+
+
+            var result = dbup.PerformUpgrade();
+
+            if (!result.Successful)
+            {
+                Console.Error.WriteLine(result.Error);
+                Environment.Exit(-1);
+            }
+        }
+
+        static void ExecuteWithRetry(Action action, Func<Exception, bool> retryCondition, Action<Exception, int, int, int> onRetry, int maxAttempts, int delay)

```

Comment:
I would recommend using Polly for retries

Reply:
I agree that Polly is the go-to library for this kind of things. I didn't want to introduce yet another third-party library to keep it simple.